classes/module/Module.php

```diff
@@ -2799,7 +2792,13 @@ public function addOverride($classname)
 
             $module_file = file($path_override);
             $module_file = array_diff($module_file, array("\n"));
-            eval(preg_replace(array('#^\s*<\?(?:php)?#', '#class\s+'.$classname.'(\s+extends\s+([a-z0-9_]+)(\s+implements\s+([a-z0-9_]+))?)?#i'), array(' ', 'class '.$classname.'Override'.$uniq), implode('', $module_file)));
+            $new_override = preg_replace(array('#^\s*<\?(?:php)?#', '#class\s+'.$classname.'(\s+extends\s+([a-z0-9_]+)(\s+implements\s+([a-z0-9_]+))?)?#i'), array(' ', 'class '.$classname.'Override'.$uniq), implode('', $module_file));
+            try {
+                $parser->parse($new_override);
+            } catch (PhpParser\Error $e) {
+                throw new Exception(sprintf(Tools::displayError('The override file "%1$s" of the module %2$s is damaged or invalid.'), 'override'.DIRECTORY_SEPARATOR.$path, $this->displayName));
+            }
+            eval($new_override);
```

Comment:
This may be overkill but why not.
However, this extra check needs to be tested properly. The documentation says eval() always returns null by default.